### DISCUSSION TITLE
On-demand-revalidation logs

### DISCUSSION THREAD

**Discussion by aadilghani1** [vercel/next.js#42070] [Ideas]
### Describe the feature you'd like to request

Currently there is no way to debug revalidation with On-demand-revalidation. It is relatively difficult to find out if the page is really not being revalidated or is there some other service causing a revalidation failure. 
Specifically in my case while building a preview setup with Sanity, We tried to hook up On-demand-revalidation here with a custom sanity webhook. Here we receive POST requests from the webhook upon any cms change onto our next.js api endpoint which does the revalidation. Due to cdn issues we were not able to use revalidate properly. The point here is, It would've been helpful if we had a debug revalidation section something similar to function logs where we can see the revalidation happening for different static pages.

### Describe the solution you'd like

We can have a section where we show the revalidation logs for every SSG page. Something similar to function logs for API routes.

### Describe alternatives you've considered

We can manually test this by adding logs in the revalidation endpoint but again, There is no way to see why res.revalidate() is failing or any way to back track this.

**Comment by AntonioChagoyaRecteq** [comment]
I have the exact same issue, also integrating Sanity Webhooks, it is very hard to test if revalidation api actually worked

Not to mention usage logs don't show uncached request anymore
